### DISCUSSION TITLE
Support for Azure Service Bus

### DISCUSSION THREAD

**Discussion by michaelegner** [microcks/microcks#928] [Q&A] [answered]
I'm currently doing a PoC for integrating microcks in our current project (spring boot based microservices). This worked out quite well for the HTTP / synchronous way due to the newly announced Testcontainers integration. We are using Azure Service Bus for async service to service communication and the implementation is based on the Java Azure SDK. Are there any plans to provide a binding for Azure Service Bus in microcks? Or should the AMQP binding be used in this case?

**Comment by lbroudoux** [comment, marked as answer]
Hi Michael,

Currently, Azure Service Bus support is not on the roadmap because you're the first one to ask for it 😉 
We always wait for the user's demand before introducing new binding support so that we are sure to have some "champion" to validate it and use it later on. Regarding AMQP support, we currently support AMQP 0.9 (aka RabbitMQ) but not the AMQP 1.0 variant. I have never used Azure Service Bus before, so I don't know what version of the protocol it uses underhood... 

One note regarding Testcontainers: our Testcontainers integration doesn't currently support AsyncAPI mocking/testing. We started with synchronous protocols but need further feedback from the community and enhancements to add the asynchronous parts.

In a nutshell, I'd say that we're definitely looking forward to adding new connectors and features! Adding a new binding usually takes 5 to 10 days of work, but we want to ensure we'll have that "champion" that sticks around. Have you played or tried some other bindings to validate that Microcks will fit your need on that part?

I'd happily connect for a quick call if you'd like to discuss it further. Let me know.

**Reply by michaelegner** [reply]
Hi Laurent,

many thanks for the quick response. 
As stated [here](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-overview#amqp-10-and-service-bus) Azure Service Bus is using AMQP 1.0 underhood. 
I will try out the RabbitMQ binding to see if Microcks fits our use case and will come back to you with the results.
